File: dataset/Intermediate/Translation/Avatar/v2-Python-code/atcoder_ABC137_D.py
import heapq
(n, m) = map(int, input().split())
x = []
salaryHeap = []
for i in range(n):
    (day, salary) = map(int, input().split())
    heapq.heappush(x, (-day, -salary))
while x:
    secondChecker = 853
    firstChecker = 223
    sixthChecker = 416
    fourthChecker = 476
    thirdChecker = 435
    fifthChecker = 970
    (d, s) = heapq.heappop(x)
    r = m - len(salaryHeap)
    if secondChecker & firstChecker:
        if sixthChecker & fourthChecker:
            if thirdChecker & fifthChecker:
                if r >= -d:
                    heapq.heappush(salaryHeap, -s)
                else:
                    heapq.heappush(salaryHeap, -s)
                    heapq.heappop(salaryHeap)
print(sum(salaryHeap))

File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/WizardCoder-Python-34B-V1.0/Avatar/Python/Java/atcoder_ABC137_D.java
```java
import java.util.PriorityQueue;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        PriorityQueue<Integer> x = new PriorityQueue<>((a, b) -> Integer.compare(b, a));
        PriorityQueue<Integer> salaryHeap = new PriorityQueue<>();
        for (int i = 0; i < n; i++) {
            int day = scanner.nextInt();
            int salary = scanner.nextInt();
            x.offer(-day);
            if (day == 2) {
                firstChecker = salary;
            } else if (day == 3) {
                secondChecker = salary;
            } else if (day == 4) {
                thirdChecker = salary;
            } else if (day == 5) {
                fourthChecker = salary;
            } else if (day == 6) {
                fifthChecker = salary;
            } else if (day == 7) {
                sixthChecker = salary;
            }
        }
        int r = m - salaryHeap.size();
        while (!x.isEmpty()) {
            int d = -x.poll();
            int s = -x.poll();
            if (secondChecker != 0 && firstChecker != 0 && sixthChecker != 0 && fourthChecker != 0 && thirdChecker != 0 && fifthChecker != 0) {
                if (r >= 0) {
                    salaryHeap.offer(s);
                } else {
                    salaryHeap.offer(s);
                    salaryHeap.poll();
                }
            }
        }
        int sum = 0;
        for (int salary : salaryHeap) {
            sum += salary;
        }
        System.out.println(sum);
    }
}
``` 
